libpod/options.go

```diff
@@ -1342,14 +1343,45 @@ func WithRootFS(rootfs string) CtrCreateOption {
 		if ctr.valid {
 			return define.ErrCtrFinalized
 		}
-		if _, err := os.Stat(rootfs); err != nil {
+
+		rootfsWithOverlay, err := withRootFSOverlay(ctr, rootfs)
+		if err != nil {
 			return err
 		}
+		if rootfsWithOverlay != "" {
+			rootfs = rootfsWithOverlay
+		}
+		if _, err = os.Stat(rootfs); err != nil {
+			return err
+		}
+
 		ctr.config.Rootfs = rootfs
 		return nil
 	}
 }
 
+func withRootFSOverlay(ctr *Container, rootfs string) (string, error) {
+	// Note: This is just an experiment and will be more generic in final draft.
+	parts := strings.SplitN(rootfs, ":", 2)
+	// lookout for --rootfs /some/path:O
+	if len(parts) > 1 && parts[1] == "O" {
+		overlaySrc := parts[0]
+		// remove tmp from here and take value of --tmpfs from cli
+		overlayDest := ctr.runtime.store.GraphRoot()
+		contentDir, err := overlay.GenerateStructure(ctr.runtime.store.GraphRoot(), ctr.ID(), "rootfs", ctr.RootUID(), ctr.RootGID())
+		if err != nil {
+			return "", errors.Wrapf(err, "rootfs-overlay: failed to create TempDir in the %s directory", overlayDest)
+		}
+		overlayMount, err := overlay.Mount(contentDir, overlaySrc, overlayDest, ctr.RootUID(), ctr.RootGID(), ctr.runtime.store.GraphOptions())
+		if err != nil {
+			return "", errors.Wrapf(err, "rootfs-overlay: creating overlay failed %q", overlaySrc)
+		}
+
+		return overlayMount.Source, nil
+	}
+	return "", nil
```

Comment:
Return rootfs here.

Reply:
Who is supposed to unmount the overlay.Mount?

Reply:
@rhatdan mount path will be removed  as soon as the  container is deleted since we are mounting at `<base(graphroot)>+ containers/storage/+ overlay-containers`. But I can explicit clean up if that is needed.

Reply:
If it gets cleaned up, then fine. I just want  you to make sure.

mount | wc
podman run ... echo hi
mount | wc

Reply:
@rhatdan verified with `mount | wc` and rootless mode cant see any dangling mounts

Reply:
@rhatdan for the first comment,  `"", nil` is returned so caller will automatically check if string is empty it will fallback to original rootfs.

Reply:
@rhatdan PTAL

Reply:
If you always return rootfs, then the caller will not need to check.

Reply:
Sure let me fix this.

Reply:
@rhatdan fixed in latest commit.